### PR TITLE
Add NormalizeLocales task

### PR DESCRIPTION
**Pull request by jasonleenaylor**
* Add task to help work with Crowdin locales
* Add MockBuildEngine to support testing
* Add task into the documentation

---
This change is [Reviewable](https://reviewable.io/reviews/sillsdev/sil.buildtasks/25)

### CODE REVIEW

File: SIL.BuildTasks.Tests/NormalizeLocalesTests.cs [hunk 1]
```diff
@@ -0,0 +1,131 @@
...
+		[Test]
+		public void DoesntCrashWhenNoNameChange()
+		{
+			FileSystemSetup(new[] { "de", "en-US" });
+
+			_task.Execute();
+
+			// Verify that the already-normalized locale is still normalized
+			VerifyLocale("de", "never-existed");
+			// Verify that the locale that needed to be normalized has been
```
Comment (ermshiperete) [inline SIL.BuildTasks.Tests/NormalizeLocalesTests.cs]: is there a word missing? `...has been`?

**Review by papeh — COMMENTED** [review]
This code was designed for translations exported to
`/%locale%/**/filename.%locale%.ext`
however, the present `translation: "Chorus.%locale%.xlf"` will automatically export to
`/%normalized_locale%/%original_path%/Chorus.%locale%.xlf`
which this code will happily not normalize.

We'll need something similar to
```
foreach (var locale in locales.Where(l => !l.Contains('-')))
    // navigate the directory tree and strip region codes
```
(no longer a need to handle Chinese specifically).

> *[Reviewable](https://reviewable.io/reviews/sillsdev/sil.buildtasks/25#-:-M4u6XXO-Zh-3CUC2442:b-g4jku8)* status: 0 of 5 files reviewed, all discussions resolved (waiting on @ermshiperete)

**Comment by jasonleenaylor** [comment]
@papeh Have you looked at the settings in Crowdin for the SIL Common Libraries recently? I think I already changed it to what this code expects so we can use the same task in both FieldWorks and the Common libraries. I intend to use this *before* generating the NuGet packages.

**Review by ermshiperete — CHANGES_REQUESTED** [review]
:lgtm:

Should add an entry to CHANGELOG.md though.

And it would be good to add `+semver:minor` to the git commit message.

File: SIL.BuildTasks.Tests/NormalizeLocalesTests.cs (hunk 1 above)
Comment (jasonleenaylor) [reply SIL.BuildTasks.Tests/NormalizeLocalesTests.cs]: It is grammatical in English. But I'll change it to make it parse easier. :)

**Review by ermshiperete — APPROVED** [review]
:lgtm:

> Reviewed 2 of 5 files at r1, 2 of 2 files at r3.
> *[Reviewable](https://reviewable.io/reviews/sillsdev/sil.buildtasks/25#-:-M4ya03g1MusW3XXiNxJ:bnfp4nl)* status: 4 of 6 files reviewed, all discussions resolved

**Pull request merged**